src/runtime_src/tools/xclbin/FDTNode.cxx

```diff
@@ -178,10 +178,17 @@ FDTNode::FDTNode(const boost::property_tree::ptree& _ptDTC,
       XUtil::TRACE(XUtil::format("Property Found: %s", keyName.c_str()).c_str());
       FDTProperty * pProperty = new FDTProperty(iter, _propertyNameFormat);
       m_properties.push_back(pProperty);
-    } else {
+    } else if (!iter->second.empty()){
       XUtil::TRACE(XUtil::format("Node Found: %s", keyName.c_str()).c_str());
       FDTNode * pFDTNode = new FDTNode(iter->second, keyName, _propertyNameFormat);
       m_nestedNodes.push_back(pFDTNode);
+    } else {
```

Comment:
If I read this code correctly, you are desiring for all unknown variables to be strings.   For this schema, it is important to follow what is being requested by the Platform team.   Instead, variables should be defined in FDTProperty::PropertyNameFormat vector.  Please do so and remove the changes in this file.

Reply:
removed